Makefile_plugin.common

```diff
@@ -0,0 +1,24 @@
+CXX = $(shell yosys-config --cxx)
```

Comment:
Documentation please?

Reply:
Should also mention how tests are invoked in the documentation?

Reply:
This is not the documentation I was expecting.  I was looking for:
 - What this template is for?
 - How to use it?
 - How to test with it?

Reply:
@litghost How about now?

Reply:
I want a description of how a developer can add automated test cases, e.g. make a directory called tests, consult this other makefile template, etc.

I think the idea here is that if a new developer comes in, how do they add a plugin without reading all the makefile itself.

I think showing an example directory structure for the makefile example plus referencing the testing makefile template will finish this off.

Reply:
Mentioned the Makefile_test.common and added an example directory structure.